tests/Microsoft.Identity.Test.Unit/BrokerTests/WamTests.cs

```diff
@@ -1,698 +1,755 @@
+                requestParams.Account = new Account(
+                   $"{TestConstants.Uid}.{TestConstants.Utid}",
+                   TestConstants.DisplayableId,
+                   null,
+                   new Dictionary<string, string>() { { TestConstants.ClientId, "wam_id_1" } }); // account has wam_id!
+
+                var accountPicker = Substitute.For<IAccountPicker>();
+
+                _accountPickerFactory.Create(Arg.Any<IntPtr>(), null, null, null, false).ReturnsForAnyArgs(accountPicker);
+
+                var wamAccountProvider = new WebAccountProvider("id", "user@contoso.com", null);
+                accountPicker.DetermineAccountInteractivelyAsync().Returns(Task.FromResult(wamAccountProvider));
+
+                var webTokenRequest = new WebTokenRequest(wamAccountProvider);
+                _aadPlugin.CreateWebTokenRequestAsync(
+                    wamAccountProvider,
+                    requestParams,
+                    isForceLoginPrompt: true,
+                     isInteractive: true,
+                     isAccountInWam: false)
+                    .Returns(Task.FromResult(webTokenRequest));
+
+                var webTokenResponseWrapper = Substitute.For<IWebTokenRequestResultWrapper>();
+                webTokenResponseWrapper.ResponseStatus.Returns(WebTokenRequestStatus.Success);
+                var webTokenResponse = new WebTokenResponse();
+                webTokenResponseWrapper.ResponseData.Returns(new List<WebTokenResponse>() { webTokenResponse });
+
+                _wamProxy.RequestTokenForWindowAsync(Arg.Any<IntPtr>(), webTokenRequest).
+                    Returns(Task.FromResult(webTokenResponseWrapper));
+                _aadPlugin.ParseSuccesfullWamResponse(webTokenResponse).Returns(_msalTokenResponse);
+
+                var atiParams = new AcquireTokenInteractiveParameters();
+
+                // Act
+                var result = await _wamBroker.AcquireTokenInteractiveAsync(requestParams, atiParams)
+                    .ConfigureAwait(false);
+
+                // Assert 
+                Assert.AreSame(_msalTokenResponse, result);
+            }
+        }
+
+        [TestMethod]
+        public async Task RemoveAADAccountAsync()
+        {
+            string aadHomeAccId = $"{TestConstants.Uid}.{TestConstants.Utid}";
+            // Arrange
+            using (var harness = CreateTestHarness())
+            {
+                var wamAccountProvider = new WebAccountProvider("id", "user@contoso.com", null);
+                var requestParams = harness.CreateAuthenticationRequestParameters(TestConstants.AuthorityConsumerTidTenant); // MSA
+                requestParams.Account = new Account(
+                                    aadHomeAccId, // matching in on home acc id
+                                    "doesnt_matter@contoso.com", // matching is not on UPN
+                                    null); // account does not have wam_id, might be coming directly from WAM
```

Comment:
when is the wam_id included and what does it mean?

Reply:
This is described in the technical aspects of the WAM integration [here](https://identitydivision.visualstudio.com/DevEx/_git/AuthLibrariesApiReview/pullrequest/1972?_a=files&path=%2F%5BWindows%5D%20Wam%20Integration%2FWAM%20-%20MSA.NET%20integration%2FWAM%20integration%20-%20technical%20aspects.md)

As a summary, apps are not generally allowed to list WAM accounts, so `wam.GetAccounts()` typically returns null. But if the user logs in interactively, WAM returns an internal "wam_account_id" back to MSAL. We store this in our token cache and if the user performs `GetAccounts` we can now return this account and allow the app to perform `AcquireTokenSilent`